### PR TITLE
Add nodeadm-check-vendor PR check

### PR DESCRIPTION
**Pull request by cartermckinnon**
**Description of changes:**

Adds a PR check that ensures the `vendor` directory is kept up to date.

Also fixes the `lint` check by ignoring the `vendor` directory.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: hack/shfmt [hunk 1]
```diff
@@ -1,11 +1,18 @@
...
+# ensure these flags are equivalent to the rules in the .editorconfig
+FLAGS="--list --language-dialect auto --indent 2 --binary-next-line --case-indent --space-redirects"
+
+cd $(dirname $0)/..
 WORKDIR=$(realpath .)
+
 SHFMT_COMMAND=$(which shmft 2> /dev/null)
 if [ -z "$SHFMT_COMMAND" ]; then
-  SHFMT_COMMAND="docker run --rm -v $WORKDIR:$WORKDIR -w $WORKDIR mvdan/shfmt"
+  SHFMT_COMMAND="docker run --rm -v $WORKDIR:${WORKDIR} -w ${WORKDIR} mvdan/shfmt"
```
Comment (ndbaker1) [inline hack/shfmt:13]: why is the first `WORKDIR` left out :(

**Review by ndbaker1 — APPROVED** [review]
ty!

File: hack/shfmt (hunk 1 above)
Comment (cartermckinnon) [reply hack/shfmt]: haha done!

**Review by Issacwww — APPROVED** [review]
ty! need this to fix fmt and lint for my PR as well hh

**Pull request merged**